backend/app/assets/javascripts/spree/backend/variant_autocomplete.js

```diff
@@ -7,10 +7,28 @@
     });
   };
 
-  $.fn.variantAutocomplete = function(searchOptions) {
-    if (searchOptions == null) {
-      searchOptions = {};
+  /**
+    * Make the element a select2 dropdown used for finding Variants. By default, the search term will be
+    * passed to the defined Spree::Config.variant_search_class by the controller with its defined scope.
+    * @param  {Object|undefined|null} options Options
+    * @param  {Function|undefined} options.searchParameters Returns a hash object for params to merge on the select2 ajax request
+    *                                                       Accepts an argument of the select2 search term. To use Ransack, define
+    *                                                       variant_search_term as a falsy value, and q as the Ransack query. Note,
+    *                                                       you need to ensure that the attributes are allowed to be Ransacked.
+    */
+  $.fn.variantAutocomplete = function(options) {
+    function extraParameters(term) {
+      if (typeof(options) === 'object') {
+        if (typeof(options['searchParameters']) === 'function') {
+          return options['searchParameters'](term)
+        } else {
+          console.warn("Solidus deprecation: Passing an object of parameters to variantAutocomplete is deprecated. Instead, on the options object, please declare `searchParameters` as a function returning the parameters.")
```

Comment:
Nice, what about adding a usage example of the new syntax? I think it might help a lot.

Reply:
Done @kennyadsl 
<img width="1563" alt="Screenshot 2022-12-22 at 10 08 31" src="https://user-images.githubusercontent.com/76776099/209099390-4131da3d-41f5-40a2-a132-2d3b377a6c24.png">